flink-runtime/src/test/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStoreTest.java

```diff
@@ -954,4 +1212,38 @@ public void testRemoveAllHandlesShouldRemoveAllPaths() throws Exception {
 
         assertThat(zkStore.getAllHandles(), is(empty()));
     }
+
+    @Test
+    public void testGetAllHandlesWithMarkedForDeletionEntries() throws Exception {
```

Comment:
hmm this seems inconsistent with what I did in k8s implementation 🤔

so if I understand it correctly, we always want `getAllHandles` to return everything

Reply:
Good point. That's the shortcoming of the `ZooKeeperStateHandleStore` not collecting the handles in a transaction. But we concluded that it's not necessary to exclude the deleted once because it can always happen getting the handles and before utilizing them that the underlying handle was deleted.

I mentioned that already in the `ZooKeeperStateHandleStore` implementation but extended the `StateHandleStore` JavaDoc accordingly. With the interface JavaDoc changes, it should be ok to have this being implemented more strictly in the k8s implementation

Reply:
Sounds good, this seems to be only used by the `SessionDispatcherLeaderProcess` and the `StateHandleStore` itself